packages/contracts-bedrock/scripts/autogen/generate-semver-lock/main.go

```diff
@@ -76,16 +76,58 @@ func processFile(file string) (*SemverLockResult, []error) {
 		return nil, []error{fmt.Errorf("failed to read artifact: %w", err)}
 	}
 
+	var sourceFilePath, contractName, contractKey string
+	for path, name := range artifact.Metadata.Settings.CompilationTarget {
+		sourceFilePath = path
+		contractName = name
+		contractKey = sourceFilePath + ":" + name
+		break
+	}
+
 	// Only apply to files in the src directory.
-	sourceFilePath := artifact.Ast.AbsolutePath
 	if !strings.HasPrefix(sourceFilePath, "src/") {
 		return nil, nil
 	}
 
-	// Check if the contract uses semver.
-	semverRegex := regexp.MustCompile(`custom:semver`)
-	semver := semverRegex.FindStringSubmatch(artifact.RawMetadata)
-	if len(semver) == 0 {
+	// Check if the contract has a version function or variable with @custom:semver tag
+	hasSemverTag := false
+	for _, node := range artifact.Ast.Nodes {
+		if node.NodeType == "ContractDefinition" && node.Name == contractName {
```

Comment:
[nit] can negate this if and use `continue` to avoid deep nesting

Reply:
Done in https://github.com/ethereum-optimism/optimism/pull/14722/commits/2ba7350fd9e28adc6335ced8db8ffd96685dcfcd